packages/admin-next/dashboard/src/hooks/table/query-v2/use-promotion-table-query.tsx

```diff
@@ -0,0 +1,32 @@
+import { AdminGetPromotionsParams } from "@medusajs/medusa"
+import { useQueryParams } from "../../use-query-params"
+
+type UsePromotionTableQueryProps = {
+  prefix?: string
+  pageSize?: number
+}
+
+export const usePromotionTableQuery = ({
+  prefix,
+  pageSize = 20,
+}: UsePromotionTableQueryProps) => {
+  const queryObject = useQueryParams(
```

Comment:
todo: filtering and search is broken due to server errors

Reply:
Thanks @fPolic, I have a separate todo on that one. Will be tackling this in a couple of PRs down